src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedBrowserStorage.cs

```diff
@@ -0,0 +1,202 @@
+        /// Asynchronously retrieves the specified data.
+        /// </para>
+        /// <para>
+        /// The first value in the <see cref="ValueTask"/> result indicates whether the data was retrieved successfully.
+        /// </para>
+        /// </summary>
+        /// <param name="purpose">A string that defines a scope for the data protection. The protected data can only be unprotected if the same purpose was previously specified when calling <see cref="SetAsync(string, string, object)"/>.</param>
+        /// <param name="key">A <see cref="string"/> value specifying the name of the storage slot to use.</param>
+        /// <returns>A <see cref="ValueTask"/> representing the completion of the operation.</returns>
+        public async ValueTask<(bool success, T result)> TryGetAsync<T>(string purpose, string key)
+        {
+            var protectedJson = await GetProtectedJsonAsync(key);
+
+            return protectedJson == null ? (false, default!) : (true, Unprotect<T>(purpose, protectedJson));
+        }
+
+        /// <summary>
+        /// <para>
+        /// Asynchronously retrieves the specified data.
+        /// </para>
+        /// <para>
+        /// If no slot with the given <paramref name="key"/> exists, the <see langword="default" /> value for <typeparamref name="T"/> is returned.
+        /// </para>
+        /// <para>
+        /// Since no data protection purpose is specified with this overload, the purpose is derived from <paramref name="key"/> and the store name. This is a good default purpose to use if the keys come from a fixed set known at compile-time.
+        /// </para>
+        /// </summary>
+        /// <param name="key">A <see cref="string"/> value specifying the name of the storage slot to use.</param>
+        /// <returns>A <see cref="ValueTask"/> representing the completion of the operation.</returns>
+        public ValueTask<T> GetValueOrDefaultAsync<T>(string key)
+            => GetValueOrDefaultAsync<T>(CreatePurposeFromKey(key), key);
+
+        /// <summary>
+        /// <para>
+        /// Asynchronously retrieves the specified data.
+        /// </para>
+        /// <para>
+        /// If no slot with the given <paramref name="key"/> exists, the <see langword="default" /> value for <typeparamref name="T"/> is returned.
+        /// </para>
+        /// </summary>
+        /// <param name="purpose">A string that defines a scope for the data protection. The protected data can only be unprotected if the same purpose was previously specified when calling <see cref="SetAsync(string, string, object)"/>.</param>
+        /// <param name="key">A <see cref="string"/> value specifying the name of the storage slot to use.</param>
+        /// <returns>A <see cref="ValueTask"/> representing the completion of the operation.</returns>
+        public async ValueTask<T> GetValueOrDefaultAsync<T>(string purpose, string key)
+        {
+            var protectedJson = await GetProtectedJsonAsync(key);
```

Comment:
What happens if deserializing the string produces a `null`?

Reply:
I believe that would indicate that the user stored a `null`. This was the thinking behind adding a `TryGetValue` - it would allow users to distinguish between "the key doesn't exist" and "the value associated with the key was null".